### DISCUSSION TITLE
How can I pass the argument (flags? filename?) into .exe ?

### DISCUSSION THREAD

**Discussion by yangzepa** [The-Wineskin-Project/WineskinServer#491] [Q&A]
Dear, Thanks for the nice project!

I'm thinking that I need to use the old word-processor software which was developed in windows, in my Mac.
I've made it executed with using the wineskin winery. 

I've tried to use the generated wrapper app with opening the file like "Open With..", 
however no luck: it kept sending me the message that file is not exist.

I googled various websites, that 'wine' itself gives the option of passing the argument, 
but our wrapper app seems not having it.

Do we have any method that I can pass the file path into the .exe file? 
I guess this question would be from my lack of the knowledge as the noob, please let me know.
Thanks in advance!

**Comment by emendelson** [comment]
@yangzepa - Open the Wineskin app inside your wrapper. Click on the Advanced button. Go to the Tools tab. Click on "Config Utility (winecfg)."

When the Wine Configuration window opens, go to the Drives tab. Click Add... Select a drive letter (maybe D:). Next to the Path field in the same window, click on Browse. You will see a list of folders on your Mac. Scroll down to Users; click on Users, then on your user name.

Now continue to navigate to one of the folders on your Mac, perhaps Documents. Click OK. Then back in the Wine Configuration window, click OK. When you are back in the Wineskin Advanced tab, close it and close Wineskin.

Now start up your wrapper. Make sure that a file that your word processor can open is in your Documents folder. In your word processor, the contents of your Documents folder will be listed as being on drive D:, and you should be able to open a file inside that folder.

Incidentally, which word-processor are you using?

**Reply by yangzepa** [reply]
first, Thanks for the reply!

thanks for the nice instruction, however I'm aware of the folder mapping function in winecfg.
What I meant were, how can I pass the argument or flag or filename into wrapper.app file for using as 'Open With' in Mac?
we know that in windows, we can type it like "dir /w", or "notepad temp.txt" to give the argument or parameter to application.
and in Mac, not everything will be same but like 'open -a /Applications/Firefox.app/bin/something -args (some arguments)'. 

if we can give the argument to wrapper in execution state, and if the given argument could be passed into .exe file like that,
then I can maybe use the software like seamlessly. 
like double-clicking python file in finder would executing the visual studio code, 
like double-clicking .docx file in finder would executing the Microsoft word. 

and if, the file location format can be changed from Mac format into windows format automatically, 
like /user/blahblah/Documents/temp.txt  to  D:\Documents\temp.txt then, (via the virtual locations you mentioned)
files could be easily handled from outside and inside the wineskin wrapper.
(this part can be easily developed with using simple .bat file, or simple python executable. )

FYI, I'm thinking to use the hancom word editor which is (happened to be) useful in South Korea.
(almost every govenment documents format is .hwp, and hancom doesn't give any options for Mac, 
so we don't have any other options, only windows PC or VM machines like parallels and vmware.)

**Reply by emendelson** [reply]
Ah - I should have understood the question. Apologies.

First, have you tried the "Associated extension" feature in the Configuration menu? Set the extension to `hwp` and see if you can associate the wrapper with .hwp files.

Next, I use an AppleScript wrapper around the vDos emulator running WordPerfect for DOS. The AppleScript wrapper sends the name of the document file and its enclosing folder to a plist file, using the `defaults write` command. Then the Unix commands in the wrapper use `defaults read` to get the name of the file and open it. 

The AppleScript wrapper also identifies the surrounding folder of the document with a DOS drive letter, and, if necessary, creates a drive letter in the DOSdevices folder of the wrapper so that WordPerfect can open the file from that drive letter.

If this is unclear, download my vDosWPMac here and modify the AppleScript and the WineryStartupScript to suit your needs:

http://www.columbia.edu/~em36/wpdos/vdoswpmac.html

(If you get a 404 error, that is because your browser replaces http:// with https://. Change it back to http://)

EDIT: This says that the Mac OS version of Hancom is still useful. Maybe you don't need to use a wrapper at all:

https://10wontips.blogspot.com/2021/12/fix-text-spacing-jumps-in-hangul-office.html
